### ISSUE TITLE
NEON doesn't generate correct ABI information for events with type parameters

### ISSUE THREAD

**Issue by devhawk** [neo-project/neo-devpack-dotnet#120]
If the delegate is specified without type parameters, the abi.json file gets the correct ABI information

``` csharp
public delegate void AlertDelegate(string one, string two);
public static event AlertDelegate Alert;
```
``` json
{
    "name":"Alert",
    "parameters":
    [
        { "name":"one", "type":"String" },
        { "name":"two", "type":"String" }
    ],
    "returntype":"Void"
}
```

If the event is specified with a delegate using type parameters, the ABI information is missing

``` csharp
public static event Action<string, string> Alert;
```
``` json
{
    "name":"Alert",
    "parameters":
    [
    ],
    "returntype":"Unknown:"
}
```

**Comment by erikzhang** [comment]
@lightszero can you check it?

**Comment by lightszero** [comment]
i will check it

**Comment by lightszero** [comment]
@devhawk we have a unittest called UnitTest_ABI_Event.cs wtih System.Action
but it workd,can you give more info about this?

**Comment by devhawk** [comment]
I'm using the master-2.x branch, which doesn't have that test. Did we add code to support generic event delegates in the master branch? 

If this is only an issue for the 2.x branch, I'm cool with closing the issue w/o fixing it.

**Comment by lightszero** [comment]
got it.I will fix it in branch 2.x too

**Comment by lock9** [comment]
Hi @devhawk, this was fixed by #122. Can we close this? Thanks.

**Comment by devhawk** [comment]
fine by me
